packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -320,6 +320,10 @@ export class ProjectLifecycleManager {
       return
     }
 
+    // Make sure remote states in the server are reset when the project is reloaded.
+    // TODO: maybe we should also reset the server state here as well?
+    this.ctx._apis.projectApi.getRemoteStates()?.reset()
```

Comment:
This was originally invoked in the `onResetServerState` callback in the `socket-base`, implemented [here](https://github.com/cypress-io/cypress/pull/18075/files#diff-040b9d41ce1efd9534f52c69af25d3d587c3a5b9cc9cb8b39a2848ec3e7e420eR338). This is still the case, but is additionally reset here when `refreshLifecycle` is called to prevent any type of issues when switching testing types, browsers mid run, etc...